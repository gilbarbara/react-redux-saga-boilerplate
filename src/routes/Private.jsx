import React from 'react';
import styled from 'styled-components';

import Github from 'containers/GitHub';

import { Box, Container, Heading, Link, Paragraph, Screen, Text, utils } from 'styled-minimal';

const Header = styled.div`
  margin-bottom: ${utils.spacer(3)};
  text-align: center;
`;

const PrivateWrapper = styled(Screen)``;

const Private = () => (
  <PrivateWrapper key="Private">
    <Container verticalPadding>
      <Header>
        <Heading>Oh hai!</Heading>
        <Paragraph>
          You can get this boilerplate{' '}
          <Link href="https://github.com/gilbarbara/react-redux-saga-boilerplate/" target="_blank">
            here
          </Link>
        </Paragraph>
      </Header>
      <Box textAlign="center" mb={4}>
        <Heading as="h5">Here's some GitHub data</Heading>
        <Text fontSize={1}>
          <i>*Just to have some requests in the sagas...</i>
        </Text>
      </Box>
      <Github />
    </Container>
  </PrivateWrapper>
);

export default Private;
