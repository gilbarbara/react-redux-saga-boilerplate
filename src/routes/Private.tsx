import React from 'react';
import styled from 'styled-components';
import { Box, Container, Heading, Link, Paragraph, Text } from 'styled-minimal';

import { spacer } from 'modules/theme';

import Github from 'containers/GitHub';

const Header = styled.div`
  margin-bottom: ${spacer(3)};
  text-align: center;
`;

function Private() {
  return (
    <Box key="Private" data-testid="Private">
      <Container ySpacing>
        <Header>
          <Heading>Oh hai!</Heading>
          <Paragraph>
            You can get this boilerplate{' '}
            <Link
              href="https://github.com/gilbarbara/react-redux-saga-boilerplate/"
              target="_blank"
            >
              here
            </Link>
          </Paragraph>
        </Header>
        <Box textAlign="center" mb={4}>
          <Heading as="h5">Here's some GitHub data</Heading>
          <Text fontSize={1}>
            <i>*Just to have some requests in the sagas...</i>
          </Text>
        </Box>
        <Github />
      </Container>
    </Box>
  );
}

export default Private;
