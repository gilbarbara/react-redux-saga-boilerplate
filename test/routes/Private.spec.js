import React from 'react';
import { mount } from 'enzyme';

import Private from 'routes/Private';

function setup() {
  const props = {
    dispatch: () => {},
    location: {},
  };

  return mount(<Private {...props} />);
}

describe('Private', () => {
  const wrapper = setup();

  it('should be a StatelessComponent', () => {
    expect(wrapper.instance()).toBeNull();
  });

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
