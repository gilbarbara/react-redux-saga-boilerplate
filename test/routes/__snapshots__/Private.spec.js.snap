// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Private should render properly 1`] = `"<div class=\\"app__private app__route\\"><div class=\\"app__container\\"></div></div>"`;
